CONTRIBUTING.md

```diff
@@ -161,6 +161,24 @@ levels, which we call "rightward drift". This makes lines shorter, making the
 code harder to read. To avoid this, try to `return` early for error cases, or
 `continue` early in a loop to skip an iteration.
 
+#### Hoist common match returns to outside match
+
+When writing a `match` expression that has arms that each share a return type
+(e.g. `Ok(...)`), hoist the commonality outside the `match`.
+
+```rust
+// Not to style
+match foo {
+    1..10 => Ok(do_one_thing()),
+    _ => Ok(do_another()),
+}
+// Better
+Ok(match foo {
+    1..10 => do_one_thing(),
+    _ => do_another(),
+})
+```
+
```

Comment:
Maybe add some explicit explanation here: "This helps separate the logic that is different per match arm from logic that is shared across arms, thus reducing code duplication."

Also I would argue this is not just for `match` but applies to `if` too -- in general, taking advantage of the idea that many syntax constructs in Rust yield an expression.

Nits for the example: add an empty line between the two sides? "Not to style" seems a slightly uncommon way of formulating this...

Reply:
Added more justification, de-emphasized the `match` context, added a space between examples and changed the comments to be less uncommon formulations.